core/controllers/topics_and_skills_dashboard.py

```diff
@@ -98,43 +98,22 @@ def get(self):
             topic_summary_dict['classroom'] = topic_classroom_dict.get(
                 topic_summary_dict['id'], None)
 
-        untriaged_skill_summary_dicts = []
         mergeable_skill_summary_dicts = []
-        categorized_skills_dict = {}
-        topics = topic_fetchers.get_all_topics()
-        for topic in topics:
-            subtopics = topic.subtopics
-            categorized_skills_dict[topic.name] = {}
-            uncategorized_skills = (
-                skill_services.get_descriptions_of_skills(
-                    topic.uncategorized_skill_ids)[0])
-            skills_list = []
-            for skill_id in topic.uncategorized_skill_ids:
-                skill_dict = {
-                    'skill_id': skill_id,
-                    'skill_description': uncategorized_skills[skill_id]
-                }
-                skills_list.append(skill_dict)
-            categorized_skills_dict[topic.name]['uncategorized'] = (
-                skills_list)
-            for subtopic in subtopics:
-                skills = (skill_services.get_descriptions_of_skills(
-                    subtopic.skill_ids))[0]
-                skills_list = []
-                for skill_id in subtopic.skill_ids:
-                    skill_dict = {
-                        'skill_id': skill_id,
-                        'skill_description': skills[skill_id]
-                    }
-                    skills_list.append(skill_dict)
-                categorized_skills_dict[topic.name][
-                    subtopic.title] = skills_list
+
+        untriaged_skill_summaries = (
+            skill_services.get_untriaged_skill_summaries(
+                skill_summaries, skill_ids_assigned_to_some_topic,
+                merged_skill_ids))
+        untriaged_skill_summary_dicts = [
```

Comment:
Minor nit: try to do these dict conversions at the very last minute, literally within the self.values.update() call at the end. Everywhere else, use the domain objects.

(Ideally the rest of this method is refactored to follow that as well, so as not to mislead other contributors. But, for this PR, I think that's optional and it's up to you whether to do that or not. Let's just not propagate the pattern.)

Reply:
done